includes/admin/class-wc-rest-payments-orders-controller.php

```diff
@@ -120,4 +138,50 @@ public function capture_terminal_payment( $request ) {
 			return new WP_Error( 'wcpay_server_error', __( 'Unexpected server error', 'woocommerce-payments' ), [ 'status' => 500 ] );
 		}
 	}
+
+	/**
+	 * Returns customer id from order. Create or update customer if needed.
+	 *
+	 * @param WP_REST_Request $request Full data about the request.
+	 */
+	public function create_customer( $request ) {
+		try {
+			$order_id = $request['order_id'];
+
+			// Do not process non-existing orders.
+			$order = wc_get_order( $order_id );
+			if ( false === $order ) {
+				return new WP_Error( 'wcpay_missing_order', __( 'Order not found', 'woocommerce-payments' ), [ 'status' => 404 ] );
+			}
+
+			$order_user    = $order->get_user();
+			$customer_id   = $order->get_meta( '_stripe_customer_id' );
+			$customer_data = WC_Payments_Customer_Service::map_customer_data( $order );
+
+			if ( ! $customer_id && $order_user ) {
+				$customer_id = $this->customer_service->get_customer_id_by_user_id( $order_user->ID );
+			}
+
+			if ( ! $order_user ) {
+				$order_user = new WP_User();
+			}
+			if ( ! $customer_id ) {
+				$customer_id = $this->customer_service->create_customer_for_user( $order_user, $customer_data );
+			} else {
+				$customer_id = $this->customer_service->update_customer_for_user( $customer_id, $order_user, $customer_data );
+			}
```

Comment:
May avoid use the negative logic checking here? And simplify it with the ternary? Something like this:

```suggestion
			$customer_id = $customer_id
				? $this->customer_service->update_customer_for_user( $customer_id, $order_user, $customer_data )
				: $this->customer_service->create_customer_for_user( $order_user, $customer_data );
```